File: 백준/Bronze/2884. 알람 시계/알람 시계.swift
let a = readLine()!
let b = a.split(separator: " ")
var x: Int = Int(b[0])!
var y: Int = Int(b[1])!

y -= 45
if y < 0 {
    y = 60 - (-y)
    x -= 1
}
if x < 0 {
    x = 24 - (-x)
}
print("\(x) \(y)")

File: 백준/Bronze/2884. 알람 시계/알람 시계.py
hour, min = map(int, input().split())

if hour > 0 and min >= 45 :
    print(hour, min - 45)
elif hour > 0 and min < 45 :
    print(hour - 1, min + 15) 
elif hour == 0 and min >= 45 :
    print(hour, min - 45) 
else :
    print(23, min + 15) 